File: frontend/src/api/attachment.ts
import api from './index'

export interface Attachment {
  id: number
  filename: string
  original_name: string
  file_size: number
  mime_type: string
  url: string
  user_id: number
  task_id: number | null
  project_id: number | null
  created_at: string
}

export function uploadAttachment(file: File, taskId?: number | string, projectId?: number | string) {
  const formData = new FormData()
  formData.append('file', file)
  if (taskId) formData.append('task_id', String(taskId))
  if (projectId) formData.append('project_id', String(projectId))
  return api.post<Attachment>('/api/attachments/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export function getTaskAttachments(taskId: number | string) {
  return api.get<Attachment[]>(`/api/tasks/${taskId}/attachments`)
}

export function getProjectAttachments(projectId: number | string) {
  return api.get<Attachment[]>(`/api/projects/${projectId}/attachments`)
}

export function deleteAttachment(id: number | string) {
  return api.delete(`/api/attachments/${id}`)
}

File: backend/internal/model/attachment.go
package model

import "time"

type Attachment struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	FileName    string    `gorm:"size:255;not null" json:"file_name"`
	FileURL     string    `gorm:"size:500;not null" json:"file_url"`
	FileSize    int64     `gorm:"not null" json:"file_size"`
	ContentType string    `gorm:"size:100" json:"content_type"`
	UserID      uint      `gorm:"not null;index" json:"user_id"`
	User        *User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
	TaskID      *uint     `gorm:"index" json:"task_id,omitempty"`
	ProjectID   *uint     `gorm:"index" json:"project_id,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
}
